File: src/dotnet/Common/Models/ResourceProviders/Agent/KnowledgeManagementAgent.cs
using System.Text.Json.Serialization;

namespace FoundationaLLM.Common.Models.ResourceProviders.Agent
{
    /// <summary>
    /// The Knowledge Management agent metadata model.
    /// </summary>
    public class KnowledgeManagementAgent : AgentBase
    {
        /// <summary>
        /// Gets or sets the vectorization settings for the agent.
        /// </summary>
        [Obsolete("Use KnowledgeSearch property instead. This property will be removed in future versions.")]
        [JsonPropertyName("vectorization")]
        public AgentVectorizationSettings? Vectorization { get; set; }

        /// <summary>
        /// Whether the agent has an inline context and does not rely on an external resource.
        /// </summary>
        [JsonPropertyName("inline_context")]
        public bool InlineContext { get; set; } = false;

        /// <summary>
        /// Set default property values.
        /// </summary>
        public KnowledgeManagementAgent() =>
            Type = AgentTypes.KnowledgeManagement;
    }
}

File: src/python/PythonSDK/foundationallm/models/agents/knowledge_management_agent.py
"""
Encapsulates the metadata for the agent fulfilling the orchestration request.
"""
from typing import Optional
from foundationallm.models.agents import AgentBase, AgentVectorizationSettings

class KnowledgeManagementAgent(AgentBase):
    """Knowlege Management Agent metadata model."""
    vectorization: Optional[AgentVectorizationSettings] = None

